frontend/src/components/admin/ShiftCalendar/ShiftCalendar.tsx

```diff
@@ -0,0 +1,144 @@
+import FullCalendar, {
+  DateSelectArg,
+  EventChangeArg,
+  EventClickArg,
+  EventInput,
+} from "@fullcalendar/react";
+import timeGridPlugin from "@fullcalendar/timegrid";
+import interactionPlugin from "@fullcalendar/interaction";
+import moment from "moment";
+import React from "react";
+import {
+  Box,
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react";
+import colors from "../../../theme/colors";
+import "./ShiftCalendar.css";
+
+type Event = {
+  id: string;
+  start: Date;
+  end: Date;
+};
+
+const ShiftCalendar = (): React.ReactElement => {
+  const [events, setEvents] = React.useState<Event[]>([]);
```

Comment:
Nit: let's import `useState` and use it directly

```suggestion
  const [events, setEvents] = useState<Event[]>([]);
```

Reply:
Also I think `events` should be a prop passed in from the parent component (CreatePostingShifts #164) so that it can have access to the events, perform error checking, and add update the global posting context.